### PR TITLE
Fury warrior Rampage analysis

### PR DESCRIPTION
**Pull request by Maldark**
Added initial Rampage usage analysis, currently only supports Frothing Berserker.
Added T21 2P and 4P analysis
Added Rampage Cancellation analysis
Added Mastery support

### CODE REVIEW

File: src/Parser/Warrior/Fury/Modules/Features/RampageFrothingBerserker.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          minor,
+          average,
+          major,
+        },
+      } = this.suggestionThresholdsFrothingBerserker;
+      
+      // Frothing Berserker users should cast Rampage at 100 rage only
+      if(this.combatants.selected.hasTalent(SPELLS.FROTHING_BERSERKER_TALENT.id)) {
+        when(this.premature_counter / this.casts_counter).isGreaterThan(minor)
+          .addSuggestion((suggest, actual, recommended) => {
```
Comment (Pewtro) [inline src/Parser/Warrior/Fury/Modules/Features/RampageFrothingBerserker.js]: Perhaps move this ``when()`` into a constant for readability.

File: src/Parser/Warrior/Fury/Modules/Features/RampageFrothingBerserker.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            return suggest(<Wrapper>Try to cast <SpellLink id={SPELLS.RAMPAGE.id} /> at 100 rage to proc <SpellLink id={SPELLS.FROTHING_BERSERKER_TALENT.id} />.</Wrapper>)
+              .icon(SPELLS.RAMPAGE.icon)
+              .actual(`${(actual * 100).toFixed(2)}% (${this.premature_counter} out of ${this.casts_counter}) of your Rampage casts were cast below 100 rage.`)
+              .recommended(`0% is recommended`)
```
Comment (Pewtro) [inline src/Parser/Warrior/Fury/Modules/Features/RampageFrothingBerserker.js]: Please use ``formatPercentage()`` here aswell

File: src/Parser/Warrior/Fury/Modules/Items/T21_2set.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    return {
+      id: `spell-${SPELLS.WARRIOR_FURY_T21_2P_BONUS_DEBUFF.id}`,
+      icon: <SpellIcon id={SPELLS.WARRIOR_FURY_T21_2P_BONUS_DEBUFF.id} />,
+      title: <SpellLink id={SPELLS.WARRIOR_FURY_T21_2P_BONUS_DEBUFF.id} />,
+      result: (
+        <Wrapper>
+          {this.owner.formatItemDamageDone(damage)}
+        </Wrapper>
+      ),
```
Comment (Pewtro) [inline src/Parser/Warrior/Fury/Modules/Items/T21_2set.js]: no need to use Wrapper if you're merely displaying this one piece of info, just have it like you have the ``<SpellLink/>``

File: src/Parser/Warrior/Fury/Modules/Items/T21_4set.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  item() {
+    return {
+      id: `spell-${SPELLS.WARRIOR_FURY_T21_4P_BONUS_BUFF.id}`,
+      icon: <SpellIcon id={SPELLS.WARRIOR_FURY_T21_4P_BONUS_BUFF.id} />,
+      title: <SpellLink id={SPELLS.WARRIOR_FURY_T21_4P_BONUS_BUFF.id} />,
+      result: (
+        <Wrapper>
+          {this.owner.formatItemDamageDone(this.damage)}
+        </Wrapper>
+      ),
```
Comment (Pewtro) [inline src/Parser/Warrior/Fury/Modules/Items/T21_4set.js]: No need for wrapper here.

**Comment by Maldark** [comment]
Fixed all issues mentioned by @Pewtro. What doubts do you have?

**Pull request merged**